frontend/src/services/auth/session.ts

```diff
@@ -1,78 +1,94 @@
 import "server-only";
 
-import { JWTPayload, jwtVerify, SignJWT } from "jose";
+import { createPublicKey, KeyObject } from "crypto";
 import { environment } from "src/constants/environments";
-import { SessionPayload, UserSession } from "src/services/auth/types";
+import {
+  API_JWT_ENCRYPTION_ALGORITHM,
+  CLIENT_JWT_ENCRYPTION_ALGORITHM,
+  decrypt,
+  encrypt,
+  newExpirationDate,
+} from "src/services/auth/sessionUtils";
+import { SimplerJwtPayload, UserSession } from "src/services/auth/types";
 import { encodeText } from "src/utils/generalUtils";
 
 // note that cookies will be async in Next 15
 import { cookies } from "next/headers";
 
-const encodedKey = encodeText(environment.SESSION_SECRET);
+let clientJwtKey: Uint8Array;
+let loginGovJwtKey: KeyObject;
 
-// returns a new date 1 week from time of function call
-export const newExpirationDate = () =>
-  new Date(Date.now() + 7 * 24 * 60 * 60 * 1000);
-
-export async function encrypt({
-  token,
-  expiresAt,
-}: SessionPayload): Promise<string> {
-  const jwt = await new SignJWT({ token })
-    .setProtectedHeader({ alg: "HS256" })
-    .setIssuedAt()
-    .setExpirationTime(expiresAt || "")
-    .sign(encodedKey);
-  return jwt;
-}
-
-export async function decrypt(
-  sessionCookie: string | undefined = "",
-): Promise<JWTPayload | null> {
-  try {
-    const { payload } = await jwtVerify(sessionCookie, encodedKey, {
-      algorithms: ["HS256"],
-    });
-    return payload;
-  } catch (error) {
-    console.error("Failed to decrypt session cookie", error);
-    return null;
+// isolate encoding behavior from file execution
+const initializeSessionSecrets = () => {
```

Comment:
this implementation probably isn't strictly necessary, but it's a little bit cleaner than instantiating on each module load, and helps deal with situations (like our build) where the keys won't be available